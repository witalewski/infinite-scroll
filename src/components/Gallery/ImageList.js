import React, { Component } from 'react';
import styled from '@emotion/styled';
import ImageItem from './ImageItem/ImageItem';
import { placeholderHeightGenerator } from '../../utils/placeholderHeightGenerator';
import { LIST_MARGIN } from '../../global/constants';

const ImageListStyled = styled.ul`
    list-style: none;
    display: flex;
    flex-direction: column;
    flex-basis: 50%;
    padding: 0;
    margin: 0 ${LIST_MARGIN}px;
`;
export class ImageList extends Component {
  render() {
    const { list, columnWidth } = this.props;
    return (
      <ImageListStyled className="image-list">
        {list.map((image, i) => (
          <li
            className="image-list--item"
            key={`${image || 'placeholder'}-${i}`}
          >
            <ImageItem
              image={image}
              height={placeholderHeightGenerator.next() * columnWidth}
            />
          </li>
        ))}
      </ImageListStyled>
    );
  }
}
