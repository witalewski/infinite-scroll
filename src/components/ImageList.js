import React, { Component } from 'react';
import styled from '@emotion/styled';
import { ImageItem } from './ImageItem';
import { LIST_MARGIN } from '../global/constants';
import loadingGif from '../assets/loading.gif'

const ImageListStyled = styled.ul`
  .image-list--item {
    background-image: url(${loadingGif});
    background-color: lightgray;
    background-repeat: no-repeat;
    background-position: center;
    background-blend-mode: multiply;
    
    margin: ${LIST_MARGIN}px 0;
    transition: height 0.2s;
  }
`;
export class ImageList extends Component {
  render() {
    const { list, columnWidth } = this.props;
    return (
      <ImageListStyled className="image-list">
        {list.map((image, i) => (
          <li
            className="image-list--item"
            key={`${image.url || 'placeholder'}-${i}`}
            style={{
              height: columnWidth * image.placeholderHeightRatio,
            }}
          >
            <ImageItem image={image} />
          </li>
        ))}
      </ImageListStyled>
    );
  }
}
