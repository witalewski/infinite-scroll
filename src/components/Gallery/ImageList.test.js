import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockImageURLs from '../../fixtures/imageURLs.json';
import { ImageList } from './ImageList';

describe('ImageList', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ImageList list={mockImageURLs} columnWidth={300} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
