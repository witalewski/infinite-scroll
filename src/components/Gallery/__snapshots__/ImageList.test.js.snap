// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList matches snapshot 1`] = `
<ForwardRef(render)
  className="image-list"
>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/8047547e25c8cdd6bf562e4c42dfa5afb9e8bdd6.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/1e9697f176056366cd08e1c5b3cc1d4bfa1554a3.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/752cdf1654b0c32427dbb070066edff6d19e940b.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/379549639744fa22e142138817f687cfe97c7aee.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/0e97728f0f6b89a537857abb6feb52e9a4134bb7.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/c47efb2e647912a9045ffaec969a5a0c936d039a.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/9428b8808f7a1a557d8d0bcfa97bff4784734e6b.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/e3365cd1d7b2ebc4f71a64641ee81410bbffce7d.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/b731401ef700ed945177a5b50eadb847c85315b9.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={225}
      image="https://cdn.shibe.online/shibes/72e5d308c1fa15e8f3c6ceadbe846ff363e87245.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/515551501802df0bb043f791a395797a5e144f93.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/4c779def35f0ca6d99f85a40468aa038d8968f51.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/50e15ff12da0450585df2f8d9983d039311f1ab6.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/a76326e2aab82d76aaada4a5d119a0a4e6b1c068.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={399}
      image="https://cdn.shibe.online/shibes/2616df5efde79c521d28020e10cc90942dc249af.jpg"
    />
  </li>
  <li
    className="image-list--item"
  >
    <inject-ImageItem
      height={300}
      image="https://cdn.shibe.online/shibes/42eaa84a993fd3f7628e9ed4bcca2fb6ab25af66.jpg"
    />
  </li>
</ForwardRef(render)>
`;
